include/micm/process/intraphase_process_builder.hpp

```diff
@@ -0,0 +1,22 @@
+/* Copyright (C) 2022 National Center for Atmospheric Research,
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#pragma once
+
+#include <micm/system/phase.hpp>
+#include <micm/system/species.hpp>
+
+namespace micm
+{
+
+  class IntraPhaseProcessBuilder
+  {
+   private:
+   public:
+    IntraPhaseProcessBuilder& For(const Phase& phase);
+    IntraPhaseProcessBuilder& With(const Species& phase);
+    IntraPhaseProcessBuilder& Build();
```

Comment:
```suggestion
    IntraPhaseProcessBuilder& For(const Phase& phase);
    IntraPhaseProcessBuilder& Reacting(const Species& reactant);
    IntraPhaseProcessBuilder& Producing(const Species& product);
    IntraPhaseProcessBuilder& WithYield(const double yield);
    IntraPhaseProcessBuilder& WithRateConstant(const RateConstant& rate_constant);
    IntraPhaseProcessBuilder& Build();
```
This was wrong on the UML diagram

Reply:
Updated tests with these. Note that a future PR removed the cpp file